Create robot account return internal server error

<img width="864" alt="Screen Shot 2021-01-13 at 10 00 58 AM" src="https://user-images.githubusercontent.com/2696760/104415486-10884000-55ad-11eb-977f-18c99c697615.png">

`Jan 13 06:58:18 192.168.48.1 core[16729]: 2021-01-13T06:58:18Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: pq: value \"1000000000000000000\" is out of range for type integer"}]} `

--- a/src/server/middleware/security/robot2.go
+++ b/src/server/middleware/security/robot2.go
@@ -28,10 +28,9 @@ func (r *robot2) Generate(req *http.Request) security.Context {
 	if !strings.HasPrefix(name, config.RobotPrefix()) {
 		return nil
 	}
-	name = strings.TrimPrefix(name, config.RobotPrefix())
 	// The robot name can be used as the unique identifier to locate robot as it contains the project name.
 	robots, err := robot_ctl.Ctl.List(req.Context(), q.New(q.KeyWords{
-		"name": name,
+		"name": strings.TrimPrefix(name, config.RobotPrefix()),
 	}), &robot_ctl.Option{
 		WithPermission: true,
 	})

--- a/src/server/v2.0/handler/robot.go
+++ b/src/server/v2.0/handler/robot.go
@@ -15,6 +15,7 @@ import (
 	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
 	"github.com/goharbor/harbor/src/server/v2.0/models"
 	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/robot"
+	"math"
 	"regexp"
 	"strconv"
 	"strings"
@@ -254,11 +255,11 @@ func (rAPI *robotAPI) requireAccess(ctx context.Context, level string, projectID
 // more validation
 func (rAPI *robotAPI) validate(d int64, level string, permissions []*models.Permission) error {
 	if !isValidDuration(d) {
-		return errors.New(nil).WithMessage("bad request error duration input").WithCode(errors.BadRequestCode)
+		return errors.New(nil).WithMessage("bad request error duration input: %d", d).WithCode(errors.BadRequestCode)
 	}
 
 	if !isValidLevel(level) {
-		return errors.New(nil).WithMessage("bad request error level input").WithCode(errors.BadRequestCode)
+		return errors.New(nil).WithMessage("bad request error level input: %s", level).WithCode(errors.BadRequestCode)
 	}
 
 	if len(permissions) == 0 {
@@ -337,7 +338,7 @@ func isValidLevel(l string) bool {
 }
 
 func isValidDuration(d int64) bool {
-	return d >= int64(-1)
+	return d >= int64(-1) && d < math.MaxInt32
 }
 
 func isValidSec(sec string) bool {

--- a/src/server/v2.0/handler/robot_test.go
+++ b/src/server/v2.0/handler/robot_test.go
@@ -1,6 +1,7 @@
 package handler
 
 import (
+	"math"
 	"testing"
 )
 
@@ -61,6 +62,18 @@ func TestValidDuration(t *testing.T) {
 		{"duration 9999",
 			9999,
 			true,
+		},
+		{"duration max",
+			math.MaxInt32 - 1,
+			true,
+		},
+		{"duration max",
+			math.MaxInt32,
+			false,
+		},
+		{"duration 999999999999",
+			999999999999,
+			false,
 		},
 	}
 	for _, tt := range tests {

